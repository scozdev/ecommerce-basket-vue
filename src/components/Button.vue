<template>
  <button type="submit" :class="[size, color]">{{ val }}</button>
</template>

<script>
export default {
  name: "Button",
  props: {
    val: {
      type: String,
      default: "ADD PRODUCT",
    },
    size: {
      type: String,
      default: "normal",
      validator: function (value) {
        return ["normal", "full"].indexOf(value) !== -1;
      },
    },
    color: {
      type: String,
      default: "primary",
      validator: function (value) {
        return ["primary", "secondary"].indexOf(value) !== -1;
      },
    },
  },
};
</script>

<style scoped>
button {
  color: var(--c-white);
  border: 0;
  outline: 0;
  padding: 10px;
  border-radius: 2px;
  font-weight: 600;
  box-shadow: 1px 1px 8px 1px #888888;
  cursor: pointer;
}
/* width */
.normal {
  width: 150px;
}
.full {
  width: 100%;
}
/* color */
.primary {
  background: var(--c-orange);
}
.secondary {
  background: var(--c-white);
  color: var(--c-black);
}
</style>