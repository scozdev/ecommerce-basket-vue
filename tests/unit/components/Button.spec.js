import { mount, shallowMount } from '@vue/test-utils';
import Button from '@/components/Button';

describe('Button.vue', () => {
  it('should match snapshot', () => {
    const wrapper = shallowMount(Button);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('renders props.val when passed', () => {
    const val = 'new message';
    const wrapper = shallowMount(Button, {
      props: { val: val },
    });
    expect(wrapper.text()).toMatch(val);
  });

  it('Val is ADD PRODUCT by default', () => {
    const wrapper = mount(Button);
    expect(wrapper.vm.val).toBe('ADD PRODUCT');
  });

  it('props.size passes with size type', () => {
    const size = Button.props.size;
    expect(size.validator('normal')).toBe(true);
    expect(size.validator('full')).toBe(true);
    expect(size.validator('qweqwe')).toBe(false);
  });

  it('renders props.size when passed', () => {
    const val = 'new message';
    const wrapper = shallowMount(Button, {
      props: { size: val },
    });
    expect(wrapper.props().size).toMatch(val);
  });

  it('Size is normal by default', () => {
    const wrapper = mount(Button);
    expect(wrapper.vm.size).toBe('normal');
  });

  it('props.color passes with color type', () => {
    const color = Button.props.color;
    expect(color.validator('primary')).toBe(true);
  });

  it('Color is primary by default', () => {
    const wrapper = mount(Button);
    expect(wrapper.vm.color).toBe('primary');
  });
});
