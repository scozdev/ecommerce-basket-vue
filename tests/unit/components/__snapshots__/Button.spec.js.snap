// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue should match snapshot 1`] = `<button type="submit" class="normal primary">ADD PRODUCT</button>`;
